motor_controller/source/flash_settings.c

```diff
@@ -84,7 +60,10 @@ void load_settings(void) {
   firstboot_check();
 
   for (int i = 0; i < NUM_SETTINGS; i++) {
-    pending_settings[i].value = saved_settings[i].value;
+    pending_settings[i] = saved_settings[i];
+  }
+  for (int i = 0; i < NUM_SETTINGS; i++) {
```

Comment:
I'm missing something here, why the second for loop to set the value if we're copying the whole struct?

Reply:
Because this was a copy-paste from what used to be main.c:232. But I suppose it should copy the whole struct here as well. Set pending settings and then set current settings.

Reply:
I have discovered what I was missing. `pending_settings != current_settings` :p